moped-editor/src/views/projects/projectView/ProjectNameComplex.js

```diff
@@ -0,0 +1,292 @@
+    <>
+      {/* This grid contains the project name and secondary name fields, the accept/cancel icons 
+      and the badge.  Multiple widths are set for each element so that they reflow responsively 
+      as the viewport becomes more narrow. */}
+      <Grid container>
+
+        {/* Primary project name field */}
+        <Grid item xs={12} sm={6} sx={{ paddingRight: "30px" }}>
+          <TextField
+            variant="standard"
+            fullWidth
+            id="project_name"
+            label={"Project Name"}
+            type="text"
+            defaultValue={props.projectData.project_name}
+            error={primaryTitleError}
+            placeholder={
+              primaryTitleError ? "Title cannot be blank" : "Enter project name"
+            }
+            multiline={false}
+            rows={1}
+            onChange={(e) => handleProjectNameChange(e)}
+            InputLabelProps={{
+              shrink: true,
+            }}
+            InputProps={{
+              classes: {
+                input: classes.projectNameEditField,
+              },
+            }}
+          />
+        </Grid>
+
+        {/* Secondary project name field */}
+        <Grid item xs={12} sm={3} sx={{ paddingRight: "30px" }}>
+          <TextField
+            variant="standard"
+            fullWidth
+            id="secondary_name"
+            label={"Secondary Name"}
+            type="text"
+            defaultValue={props.projectData.project_name_secondary}
+            placeholder={ "Project Byline" }
+            multiline={false}
+            rows={1}
+            onChange={(e) => handleProjectNameChange(e)}
+            InputLabelProps={{
+              shrink: true,
+            }}
+            InputProps={{
+              classes: {
+                input: classes.projectSecondaryNameEditField,
+              },
+            }}
+          />
+        </Grid>
+
+        {/* Accept / Cancel icons.
+        This grid item gets a minimum width to prevent it from reflowing onto two lines. */}
+        <Grid item xs={12} sm={1} sx={{ minWidth: "65px" }}>
+          <Icon
+            className={classes.editIcons}
+            onClick={(e) => handleAcceptClick(e)}
+          >
+            check
+          </Icon>
+          <Icon
+            className={classes.editIcons}
+            onClick={(e) => handleCancelClick(e)}
+          >
+            close
+          </Icon>
+        </Grid>
+
+        {/* The status badge. Here, we're going to jog it down a bit to make it visually centered
+        along the horizontal midline of the project name input field. */}
+        <Grid item xs={12} md={2}>
+          <Box sx={{ display: "inline", position: "relative", top: "5px" }}>
+            <ProjectStatusBadge
+              phaseKey={props.currentPhase?.phase_key}
+              phaseName={props.currentPhase?.phase_name}
+            />
+          </Box>
+        </Grid>
+
+      </Grid>
+    </>
+  );
+
+  const commonJSX = (
+    <>
```

Comment:
does this need the empty tag around it, if it only has one component in it?

Reply:
No, they can go for sure. I think I have just gotten in the habit of adding them whenever I start defining JSX, but you're 100% right. https://github.com/cityofaustin/atd-moped/pull/1302/commits/85a54b2f7249d2c8f521b6c900c8ba2043e87924